swarm/storage/resource_test.go

```diff
@@ -0,0 +1,245 @@
+package storage
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/binary"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"golang.org/x/net/idna"
+
+	"github.com/ethereum/go-ethereum/contracts/ens"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	blockCount = uint64(4200)
+)
+
+func init() {
+	log.Root().SetHandler(log.CallerFileHandler(log.LvlFilterHandler(log.LvlTrace, log.StreamHandler(os.Stderr, log.TerminalFormat(true)))))
+}
+
+type FakeRPC struct {
+	blockcount *uint64
+}
+
+func (r *FakeRPC) BlockNumber() (string, error) {
+	return strconv.FormatUint(*r.blockcount, 10), nil
+}
+
+func TestResourceValidContent(t *testing.T) {
+	// privkey for signing updates
```

Comment:
I think extracting the setup code for the tests (for example from line 39 to 55) would make tests easier to read, and would make more tests easier to write.

Reply:
I tried to factor out the tests, but:

the `TestResourceHandler` test makes another handler inline in the code and thus needs access to `rpcserver` and `datadir`.

the `TestResourceValidContent` needs the `privkey` 

That means that the shared code in the end is just the `ResourceHandler` constructor. Either that, or all those variables along with the `ResourceHandler` and a shutdown-function (for removing `datadir` and closing `rpcserver`) must be returned, which isn't very clear either.

Any other ideas on how to make this less cluttered?

Reply:
I think the multiple return values are not so bad. Or you can just define a struct with all this (with a name something like `TestConfig`)